common/components/AmountFieldFactory/AmountInputFactory.tsx

```diff
@@ -11,23 +11,23 @@ interface OwnProps {
 }
 
 interface StateProps {
+  isNonStandard: boolean;
   currentValue: ICurrentValue;
-  validAmount: boolean;
 }
 
 type Props = OwnProps & StateProps;
 
 class AmountInputClass extends Component<Props> {
   public render() {
-    const { currentValue, onChange, withProps, validAmount } = this.props;
+    const { currentValue, onChange, withProps, isNonStandard } = this.props;
 
     return (
       <Query
         params={['readOnly']}
         withQuery={({ readOnly }) =>
           withProps({
             currentValue,
-            isValid: validAmount,
+            isValid: !!currentValue.value || isNonStandard,
```

Comment:
It took me a while to figure out that `currentValue.value` is null if it exceeds the balance. This is a little unintuitive IMO. What do we do to get the value field's value in the non-standard transactions?

Reply:
https://github.com/MyEtherWallet/MyEtherWallet/wiki/Transaction-Validity-Spec
^ For checking for non-standard transactions

Reply:
And yeah, i hear ya on the null value, https://github.com/MyEtherWallet/MyEtherWallet/issues/901
The initial thought process behind it was to make it impossible to send a transaction if a field was invalid ( as the transaction would have null fields and then be rejected by a node). But this lead to the inability to set custom error messages.